tests/integration/models/refunds.rs

```diff
@@ -0,0 +1,105 @@
+use crate::utils::{get_test_db_pool, random_price, random_simple_ascii_string};
+use lib::models::{
+    refunds::{PartialRefund, Refund, RefundModel},
+    status_history::{Status, UpdateStatus},
+};
+use pretty_assertions::assert_eq;
+use time::OffsetDateTime;
+use uuid::Uuid;
+
+pub fn make_fake_refund() -> Refund {
+    Refund::new(PartialRefund {
+        id: Uuid::new_v4(),
+        subscription_id: random_simple_ascii_string(),
+        refund_id: random_simple_ascii_string(),
+        refund_created: OffsetDateTime::now_utc(),
+        refund_amount: random_price(),
+        refund_status: Some(random_simple_ascii_string()),
+        refund_reason: Some(random_simple_ascii_string()),
+    })
+}
+
+pub async fn save_refund(model: &RefundModel<'_>, refund: &Refund) {
+    model
+        .create_from_refund(refund)
+        .await
+        .expect("Failed to create test object.");
+}
+
+#[tokio::test]
+async fn test_refund_model_create_from_refund_struct() {
+    let db_pool = get_test_db_pool().await;
+    let model = RefundModel { db_pool: &db_pool };
+    let r = make_fake_refund();
+    save_refund(&model, &r).await;
+    let result = model
+        .fetch_one_by_refund_id(&r.refund_id)
+        .await
+        .expect("Could not fetch from DB.");
+    assert_eq!(result, r);
+}
+
+#[tokio::test]
+async fn test_refund_model_fetch_one_by_refund_id() {
```

Comment:
Is this not exactly the same as the test above?

Reply:
:D

Reply:
211f258